src/components/TimeseriesPreview/TimeseriesPreview.tsx

```diff
@@ -25,6 +25,8 @@ export interface TimeseriesPreviewProps {
   timeseriesId: number;
   color?: string;
   dateFormat?: string;
+  customName?: ((name?: string) => string) | string;
```

Comment:
providing `string` type here is redundant. Better to have just function:

```typescript
(name?: string) => string
```

better to rename – `nameFormatter`